### ISSUE TITLE
Save plots in other formats, vectorial and image

### ISSUE THREAD

**Issue by raultomasmora** [pycaret/pycaret#1610]
**Is your feature request related to a problem? Please describe.**
Currently it is only possible to save plots in png format, but there is the possibility to implement other output files (pdf, svg, eps, jpg, tif).

**Describe the solution you'd like**
Possibility to save plots in other formats, such as vector (pdf, svg, eps) or image (jpg, tif, png).

**Describe alternatives you've considered**
I have made an example to demonstrate that it is possible to create vector plots from a pycaret model. It uses yellowbrick, the same package that pycaret uses. The implementation seems to be very easy, although I am not able to program it.

**Additional context**
Attached is an example of how vector plots (pdf, svg, eps) can be derived from pycaret models, using yellowbrick. It has been tested with googlecolab and pycaret 2.3.4.

#---- data
from pycaret.datasets import get_data
dataset = get_data('diamond')

#---- model pycaret
from pycaret.regression import *
exp_reg101 = setup(data = dataset, target = 'Price', session_id=123, 
                   remove_multicollinearity = True, multicollinearity_threshold = 0.95) 

lightgbm = create_model('lightgbm')

#---- plot
from yellowbrick.regressor import ResidualsPlot

X_train, X_test, y_train, y_test = get_config('X_train'), get_config('X_test'), get_config('y_train'), get_config('y_test')

visualizer = ResidualsPlot(lightgbm) # regresión lineal
visualizer.fit(X_train, y_train) # Fit the training data to the visualizer
visualizer.score(X_test, y_test) # Evaluate the model on the test data
visualizer.poof(outpath="ResidualsPlot.pdf") # VECTOR .pdf .eps .svg, RASTER .png .jpg .tif
#visualizer.poof()  # Finalize and render the figure

#---- file extensions supported
import matplotlib.pyplot as plt
plt.figure().canvas.get_supported_filetypes()

**Comment by moezali1** [comment]
@raultomasmora I like this idea. Would you like to work on this enhancement?

**Comment by raultomasmora** [comment]
I am not a programmer or computer scientist, I am a researcher in construction and housing market, interested in machine learning. I don't feel qualified to program it, I only know that I can print the plots in pdf and I wanted to share it with others. It would be very useful to implement it in pycaret.

**Comment by sorenwacker** [comment]
The plots that are saved are also lacking legend and labels etc. It would be great if e.g. a matplotlib figure would be returned instead, so the plots could be customized and saved in arbitrary formats. Also a method to change the filename would be great. 

So, I really appreciate your code here. However, that means for each kind of plot we need a lot of overhead to create the plot etc.

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
